### DISCUSSION TITLE
Extract Port Field with PyAEDT

### DISCUSSION THREAD

**Discussion by ptrem** [ansys/pyaedt#3037] [General]
Hi,
I would like to know if there is a possibility to export the port fields with PyAEDT.
So far I use the [export_field_file_on_grid](https://aedt.docs.pyansys.com/version/stable/API/_autosummary/pyaedt.modules.AdvancedPostProcessing.PostProcessor.export_field_file_on_grid.html#pyaedt.modules.AdvancedPostProcessing.PostProcessor.export_field_file_on_grid) to export the field data on a defined grid.
Now I would like to export the port fields like it is possible in HFSS.

Is there a function to directly export the port fields or at least a work around to extract the data?

Thank you

![MicrosoftTeams-image](https://github.com/ansys/pyaedt/assets/58425593/a5201369-10bb-42c0-afe2-07da767298e3)

**Comment by Samuelopez-ansys** [comment]
@ptrem You can use [plot_field ](https://aedt.docs.pyansys.com/version/dev/API/_autosummary/pyaedt.modules.AdvancedPostProcessing.PostProcessor.plot_field.html#plot-field )method, you need to pass the object name and the field you want to export.

ANother option is using https://aedt.docs.pyansys.com/version/dev/API/_autosummary/pyaedt.modules.AdvancedPostProcessing.PostProcessor.create_fieldplot_surface.html
